src/libraries/System.Collections/src/System/Collections/Generic/PriorityQueue.cs

```diff
@@ -301,6 +293,14 @@ public void EnqueueRange(IEnumerable<(TElement Element, TPriority Priority)> ite
             }
             else
             {
+                if (EnumerableHelpers.TryGetCount(items, out int count) && count > 0)
+                {
+                    if (_nodes.Length < _size + count)
```

Comment:
```suggestion
                if (EnumerableHelpers.TryGetCount(items, out int count) && count > 0)
                {
                    if (_nodes.Length - count < _size)
```

Reply:
(Also there's not really a need for the nested ifs: it could all be one condition)